frontend/src/lib/api.ts

```diff
@@ -3348,17 +3348,49 @@ const api = {
 
     dataWarehouse: {
         async total_rows_stats(options?: ApiMethodOptions): Promise<{
-            billingAvailable: boolean
-            billingInterval: string
-            billingPeriodEnd: string
-            billingPeriodStart: string
-            materializedRowsInBillingPeriod: number
-            totalRows: number
-            trackedBillingRows: number
-            pendingBillingRows: number
+            billing_available: boolean
+            billing_interval: string
+            billing_period_end: string
+            billing_period_start: string
+            materialized_rows_in_billing_period: number
+            total_rows: number
+            tracked_billing_rows: number
+            pending_billing_rows: number
         }> {
             return await new ApiRequest().dataWarehouse().withAction('total_rows_stats').get(options)
         },
+
+        async recentActivity(options?: ApiMethodOptions & { limit?: number; offset?: number }): Promise<{
+            results: Array<{
```

Comment:
There's a `PaginatedResponse` type that is pretty standardized for all paginatable APIs. Should change this to match that. Also add a type for the result objects themselves